VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -154,11 +154,11 @@ private void setCurrentUnit(CourseComponent component){
 
         environment.getDatabase().updateAccess(null, selectedUnit.getId(), true);
 
-        CourseComponent parent = component.getParent();
+        CourseComponent parentSectionComponent = component.getParent().getParent();
         String prefName = PrefManager.getPrefNameForLastAccessedBy(getProfile()
             .username, selectedUnit.getCourseId());
         final PrefManager prefManager = new PrefManager(MainApplication.instance(), prefName);
-        prefManager.putLastAccessedSubsection(parent.getId(), false);
+        prefManager.putLastAccessedSubsection(parentSectionComponent.getId(), false);
```

Comment:
@miankhalid: If the user switches the section from the `CourseUnitNavigationActivity`, then the back stack of `CourseOutlineActivity` is automatically recreated to support the new navigation upon pressing back. Still, if we are going to store the full path as @aleffert suggested then we can revert to the original logic here.